cmd/clusterctl/client/cluster/crd_migration.go

```diff
@@ -129,9 +126,8 @@ func (m *crdMigrator) run(ctx context.Context, newCRD *apiextensionsv1.CustomRes
 	// This way we can make sure that all CR objects are now stored in the current storage version.
```

Comment:
Some parts of this godoc seem oudated.

I would suggest:
* remove l.122+123
* move l.124 to the end of the godoc block